implementations/rust/ockam/ockam_command/src/node/mod.rs

```diff
@@ -81,82 +79,3 @@ pub struct NodeOpts {
     #[arg(global = true, id = "at", value_name = "NODE_NAME", long)]
     pub at_node: Option<String>,
 }
-
-/// If the required node name is the default node but that node has not been initialized yet
-/// then initialize it
-pub fn initialize_node_if_default(opts: &CommandGlobalOpts, node_name: &Option<String>) {
```

Comment:
_Observation_

We'll lose some of the default behaviors if we remove this (unless I'm missing its replacement), but I can see how removing it helps simplify the changes done in the command layer.

Reply:
You're correct. The default node data is always created and stored when you need it but the default node itself is not started automatically. The reasons for doing this are:

 - that simplifies the calls as you mention
 - in most (all?) the scripts I've seen we create a node first before creating an outlet, or another resource on it. In other places, where we get information doing, `ockam credential get` with a default node created in the background makes little sense
 - since spawning a default creates a resource that takes a real process on a machine I think it is better to be explicit about what we want to be running or not
 
 Please tell me if I am missing a scenario where the default spawning of a node is really necessary.